shotover-proxy/src/transforms/cassandra/sink_cluster.rs

```diff
@@ -31,29 +34,52 @@ impl CassandraSinkClusterConfig {
         Ok(Transforms::CassandraSinkCluster(CassandraSinkCluster::new(
             self.first_contact_points.clone(),
             chain_name,
+            self.data_center.clone(),
             tls,
         )))
     }
 }
 
 pub struct CassandraSinkCluster {
     contact_points: Vec<String>,
-    outbound: Option<CassandraConnection>,
+    contact_point_connection: Option<CassandraConnection>,
+    init_handshake: Vec<Message>,
+    init_handshake_address: Option<String>,
+    init_handshake_complete: bool,
     chain_name: String,
     failed_requests: Counter,
     tls: Option<TlsConnector>,
     pushed_messages_tx: Option<mpsc::UnboundedSender<Messages>>,
+    peer_table: FQName,
+    data_center: String,
+    /// A local clone of topology_task_nodes
+    /// Internally stores connections to the nodes
+    local_nodes: Vec<CassandraNode>,
+    /// Only written to by the topology task
+    /// Transform instances should never write to this.
+    topology_task_nodes: Arc<RwLock<Vec<CassandraNode>>>,
```

Comment:
Might be a candidate for https://docs.rs/arc-swap/latest/arc_swap/.

Then we can avoid the lock/drop thing.

Reply:
oh! I was researching these other types of locks earlier.
I decided it wasnt worth blocking progress over and I could come back to it later.
I did get a bit confused trying to understand how arc_swap works earlier, but I'll take a another look when I get the chance.

Reply:
Sure, let's not block progress on it then, especially since the solution is just an extra line. I'll open an issue to investigate properly later.